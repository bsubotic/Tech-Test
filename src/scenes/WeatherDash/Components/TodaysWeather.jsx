import T from "i18n-react";
import React, { useState } from "react";
import classNames from "classnames";
import AnimateHeight from "react-animate-height";

import Button from "components/Button";
import { ReactComponent as Humidity } from "images/humidity.svg";

import Forecast from "./Forecast";

const HumidityAndWeatherIcon = ({ humidity, icon }) => (
  <div className="flex justify-between h4">
    <div className="flex pa4">
      <span className="f2 mr1">{humidity}</span>
      <div className="flex flex-column">
        <Humidity className="h1" />
        <span className="f5 mt2">%</span>
      </div>
    </div>
    <div className="">
      <img src={icon} alt="" />
    </div>
  </div>
);

const Temperature = ({ temperature }) => (
  <div className="mt4 self-center">
    <span className="f1">{temperature}</span>
    <span className="f2">Â°C</span>
  </div>
);

const PopupListForecast = ({ cityName }) => {
  const [heightEmpty, setHeightEmpty] = useState("auto");
  const [heightFull, setHeightFull] = useState(0);

  const toggle = () => {
    setHeightFull(heightFull === 0 ? "auto" : 0);
    setHeightEmpty(heightEmpty === 0 ? "auto" : 0);
  };

  return (
    <div>
      <AnimateHeight
        duration={500}
        height={heightEmpty} // see props documentation below
      >
        <div className="h4" />
      </AnimateHeight>
      <Button
        className={classNames("bg-white bg-inherit h3", {
          br2: heightFull === 0,
          "br--bottom": heightFull === 0
        })}
        text={T.translate("forecast")}
        onClick={toggle}
      />
      <AnimateHeight
        duration={500}
        height={heightFull} // see props documentation below
      >
        <div className="pa2 pt0 bg-white br2 br--bottom h5">
          <div className="bt b--off-black overflow-y-auto h5">
            <Forecast cityName={cityName} />
          </div>
        </div>
      </AnimateHeight>
    </div>
  );
};

export default ({ condition, cityName, temperature, ...humidityAndIcon }) => {
  return (
    <div className="flex flex-column min-h-100">
      <div className="flex flex-column justify-between" style={{ flexGrow: 1 }}>
        <HumidityAndWeatherIcon {...humidityAndIcon} />
        <div className="self-center f2">{condition}</div>
        <div className="self-center f3">{cityName}</div>
        <Temperature temperature={temperature} />
        <PopupListForecast cityName={cityName} />
      </div>
    </div>
  );
};
