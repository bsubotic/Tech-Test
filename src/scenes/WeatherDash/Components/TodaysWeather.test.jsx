import React from "react";
import renderer from "react-test-renderer";

import TodaysWeather from "./TodaysWeather";

test("render an enabled submit button", () => {
  const tree = renderer.create(<TodaysWeather />).toJSON();
  expect(tree).toMatchSnapshot();
});
