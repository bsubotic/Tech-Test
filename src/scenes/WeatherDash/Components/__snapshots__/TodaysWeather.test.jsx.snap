// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render an enabled submit button 1`] = `
<div
  className="flex flex-column min-h-100"
>
  <div
    className="flex flex-column justify-between"
    style={
      Object {
        "flexGrow": 1,
      }
    }
  >
    <div
      className="flex justify-between h4"
    >
      <div
        className="flex pa4"
      >
        <span
          className="f2 mr1"
        />
        <div
          className="flex flex-column"
        >
          <svg
            className="h1"
          >
            humidity.svg
          </svg>
          <span
            className="f5 mt2"
          >
            %
          </span>
        </div>
      </div>
      <div
        className=""
      >
        <img
          alt=""
        />
      </div>
    </div>
    <div
      className="self-center f2"
    />
    <div
      className="self-center f3"
    />
    <div
      className="mt4 self-center"
    >
      <span
        className="f1"
      />
      <span
        className="f2"
      >
        Â°C
      </span>
    </div>
    <div>
      <div
        aria-hidden={false}
        className="rah-static rah-static--height-auto"
        style={
          Object {
            "height": "auto",
            "overflow": "visible",
          }
        }
      >
        <div
          style={Object {}}
        >
          <div
            className="h4"
          />
        </div>
      </div>
      <button
        className="br0 pointer pa3 w-100 outline-0 bn bg-white bg-inherit h3 br2 br--bottom"
        onClick={[Function]}
      >
        forecast
      </button>
      <div
        aria-hidden={true}
        className="rah-static rah-static--height-zero"
        style={
          Object {
            "height": 0,
            "overflow": "hidden",
          }
        }
      >
        <div
          style={Object {}}
        >
          <div
            className="pa2 pt0 bg-white br2 br--bottom h5"
          >
            <div
              className="bt b--off-black overflow-y-auto h5"
            >
              <div
                className="h-100 w-100 ph2"
              >
                <div
                  className="Spinner min-h-100 relative"
                >
                  <div
                    className="loader"
                  >
                    <div
                      className="square"
                    />
                    <div
                      className="square"
                    />
                    <div
                      className="square last"
                    />
                    <div
                      className="square clear"
                    />
                    <div
                      className="square"
                    />
                    <div
                      className="square last"
                    />
                    <div
                      className="square clear"
                    />
                    <div
                      className="square "
                    />
                    <div
                      className="square last"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
